[CI] PushQueriesIT testEqualityAndOther {SEMANTIC_TEXT_WITH_KEYWORD} failing

**Build Scans:**
- [elasticsearch-intake #30750 / part3](https://gradle-enterprise.elastic.co/s/3ljjs7k7mbgpg)
- [elasticsearch-pull-request #101867 / part-3](https://gradle-enterprise.elastic.co/s/dpes7434bsjjm)
- [elasticsearch-pull-request #101641 / part-3](https://gradle-enterprise.elastic.co/s/mtiwk4yxmfwr2)
- [elasticsearch-pull-request #101568 / part-3](https://gradle-enterprise.elastic.co/s/5nmumnuabrt7c)
- [elasticsearch-pull-request #101442 / part-3](https://gradle-enterprise.elastic.co/s/edlqwhzblzodc)
- [elasticsearch-pull-request #101382 / part-3](https://gradle-enterprise.elastic.co/s/iyxtgpn3ufxcw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEqualityAndOther {SEMANTIC_TEXT_WITH_KEYWORD}" -Dtests.seed=9D9D56EBFEA80641 -Dtests.locale=om-KE -Dtests.timezone=Europe/Ljubljana -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEqualityAndOther%20%7BSEMANTIC_TEXT_WITH_KEYWORD%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
             operator: "LuceneSourceOperator[shards = [test:0], maxPageSize = 255, remainingDocs = 2147483646]"
               status: a map containing
      processed_queries: a list containing
                        0: expected ("FieldExistsQuery [field=foo]" or "foo:[1 TO 1]") but was "*:*"
       processed_slices: <1> unexpected but ok
       processed_shards: <[test:0]> unexpected but ok
           total_slices: <1> unexpected but ok
              slice_max: <0> unexpected but ok
                current: <2147483647> unexpected but ok
              slice_min: <0> unexpected but ok
           rows_emitted: <1> unexpected but ok
          pages_emitted: <1> unexpected but ok
          process_nanos: <156431> unexpected but ok
partitioning_strategies: <{test:0=SHARD}> unexpected but ok
            slice_index: <0> unexpected but ok
```

**Issue Reasons:**
- [main] 6 failures in test testEqualityAndOther {SEMANTIC_TEXT_WITH_KEYWORD} (1.5% fail rate in 405 executions)
- [main] 5 failures in step part-3 (2.2% fail rate in 230 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (2.2% fail rate in 231 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -498,9 +498,6 @@ tests:
 - class: org.elasticsearch.smoketest.SmokeTestIngestWithAllDepsClientYamlTestSuiteIT
   method: test {yaml=ingest/100_sampling_with_reroute/Test get sample with multiple reroutes}
   issue: https://github.com/elastic/elasticsearch/issues/137457
-- class: org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT
-  method: testEqualityAndOther {SEMANTIC_TEXT_WITH_KEYWORD}
-  issue: https://github.com/elastic/elasticsearch/issues/137491
 - class: org.elasticsearch.xpack.ilm.CCRIndexLifecycleIT
   method: testBasicCCRAndILMIntegration {targetCluster=FOLLOWER}
   issue: https://github.com/elastic/elasticsearch/issues/137494

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/PushQueriesIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/PushQueriesIT.java
@@ -178,10 +178,11 @@ public class PushQueriesIT extends ESRestTestCase {
             case CONSTANT_KEYWORD, MATCH_ONLY_TEXT_WITH_KEYWORD -> List.of("*:*");
             case SEMANTIC_TEXT_WITH_KEYWORD ->
                 /*
-                 * single_value_match is here because there are extra documents hiding in the index
-                 * that don't have the `foo` field.
+                 * FieldExistsQuery is because there are extra documents hiding in the index
+                 * that don't have the `foo` field. "*:*" is because sometimes we end up on
+                 * a shard where all `foo = 1`.
                  */
-                List.of("FieldExistsQuery [field=foo]", "foo:[1 TO 1]");
+                List.of("FieldExistsQuery [field=foo]", "foo:[1 TO 1]", "*:*");
         };
         ComputeSignature dataNodeSignature = switch (type) {
             case AUTO, CONSTANT_KEYWORD, KEYWORD, TEXT_WITH_KEYWORD -> ComputeSignature.FILTER_IN_QUERY;

